networkx/algorithms/operators/product.py

```diff
@@ -462,7 +463,7 @@ def rooted_product(G, H, root):
     The nodes of G and H are not relabeled.
     """
     if root not in H:
-        raise nx.NetworkXError("root must be a vertex in H")
+        raise NodeNotFound("root must be a vertex in H")
```

Comment:
Can you just use `nx.NodeNotFound` instead of importing the specific exception?
Then this only change `NetworkXError` to `NodeNotFound`.

Reply:
But really, is the type of this exception a problem?

Reply:
Thanks for Mentioning
1. Used `nx.NodeNotFound` and removed unnecessary import.

**is the type of this exception a problem?**
well, `nx.NodeNotFound` is indeed more specific and descriptive of the actual error condition. also more specific exception can make the code's intent clearer.